src/components/Mission/Mission .js

```diff
@@ -1,48 +1,79 @@
-                Join Mission
-              </button>
-            </tr>
-          </div>
-        </tr>
-        <tr>
-          <td>Jacob</td>
-          <td>Thornton</td>
-          <div className="btns">
-            <tr>
-              <button type="button" className="btn btn-info ms-3">
-                NOT A MEMBER
-              </button>
-              <button type="button" className="btn btn-outline-danger ms-3">
-                Join Mission
-              </button>
+function Mission() {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(getMissionsData());
+  }, [dispatch, getMissionsData]);
+  const missions = useSelector((state) => state.mission);
+
+  const handleJoinMission = ({ target }) => {
+    const { id } = target;
+    dispatch(JoinMission(id));
+  };
+
+  return (
+    <Container>
+      <Table striped bordered hover size="lg" className="container my-5">
+        <thead>
+          <tr>
+            <th>mission</th>
+            <th>description</th>
+            <th>dtatus</th>
+          </tr>
+        </thead>
+        <tbody>
+          {missions.map((mission) => (
+            <tr key={mission.id}>
+              <td>{mission.name}</td>
+              <td>{mission.description}</td>
+              <td>
+                {!mission.canceled ? (
+                  <div className="btns">
+                    <tr className="d-flex">
+                      <button
+                        type="button"
+                        className="btn btn-secondary text-white py-3"
+                      >
+                        NOT A MEMBER
+                      </button>
+                      <button
+                        id={mission.id}
+                        onClick={handleJoinMission}
+                        type="button"
+                        className="btn btn-outline-dark ms-3 py-3"
+                      >
+                        Join Mission
+                      </button>
+                    </tr>
+                  </div>
+                ) : (
+                  <div className="btns">
+                    <tr className="d-flex">
+                      <button type="button" className="btn btn-info text-white py-3">
+                        Active Member
+                      </button>
+                      <button
+                        id={mission.id}
+                        onClick={handleJoinMission}
+                        type="button"
+                        className="btn btn-outline-danger ms-3 px-3"
+                      >
+                        Leave Mission
+                      </button>
+                    </tr>
+                  </div>
+                )}
+              </td>
             </tr>
-          </div>
-        </tr>
-      </tbody>
-    </Table>
-  </>
-);
+          ))}
+        </tbody>
+      </Table>
+    </Container>
+  );
+}
 
-export default MyMission;
+export default Mission;
```

Comment:
As you will change the name of it please make sure to update the export also, so that it has the actual name of it.
`export default MyMission;` there's nothing else to say. Please fix this and everything will be approved.